exporter/awskinesisexporter/internal/batch/aggregator.go

```diff
@@ -0,0 +1,75 @@
+package batch
+
+import (
+	"crypto/md5"
+
+	k "github.com/aws/aws-sdk-go/service/kinesis"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	magicNumber           = []byte{0xF3, 0x89, 0x9A, 0xC2}
+	partitionKeyIndexSize = 8
+	maxAggregationCount   = 4294967295
+	maxRecordSize         = 1 << 20
+)
+
+type Aggregator struct {
+	buf    []*Record
+	pkeys  []string
+	nbytes int
+}
+
+func (a *Aggregator) IsRecordAggregative(data []byte, partitionKey string) bool {
+	return data != nil && len(data)+len([]byte(partitionKey)) <= maxRecordSize
+}
+
+func (a *Aggregator) IsBatchFull(data []byte, partitionKey string) bool {
+	nbytes := len(data) + len([]byte(partitionKey))
+	return nbytes+a.nbytes+md5.Size+len(magicNumber)+partitionKeyIndexSize > maxRecordSize || len(a.buf) >= maxAggregationCount
+}
+
+func (a *Aggregator) Put(data []byte, partitionKey string) {
+	if len(a.pkeys) == 0 {
+		a.pkeys = []string{partitionKey}
```

Comment:
By doing this, does this mean that each aggregator key will only have the partition key of the first value written?

Reply:
Yes, we could split the aggregated records by partition key but that will result in small aggregate records.
So maybe it's worth implementing this but atm we anyways use a random partition key so this is not changing the behavior of the exporter.